motoman_gp200r_support/urdf/gp200r_macro.xacro

```diff
@@ -0,0 +1,177 @@
+            </geometry>
+        </collision>
+    </link>
+    <link name="${prefix}link_6_t">
+        <visual>
+            <geometry>
+                <mesh filename="package://motoman_gp200r_support/meshes/gp200r/visual/link_6_t.stl"/>
+            </geometry>
+            <xacro:material_yaskawa_blue/>
+        </visual>
+        <collision>
+            <geometry>
+                <mesh filename="package://motoman_gp200r_support/meshes/gp200r/collision/link_6_t.stl"/>
+            </geometry>
+        </collision>
+    </link>
+    <!-- end of link list -->
+
+    <!-- joint list -->
+    <joint name="${prefix}joint_1_s" type="revolute">
+        <parent link="${prefix}base_link"/>
+        <child link="${prefix}link_1_s"/>
+        <origin xyz="0 0 0.450" rpy="0 0 0" />
+        <axis xyz="0 0 1" />
+        <limit lower="${radians(-180)}" upper="${radians(180)}" effort="7060.79" velocity="${radians(90)}"/>
+    </joint>
+    <joint name="${prefix}joint_2_l" type="revolute">
+        <parent link="${prefix}link_1_s"/>
+        <child link="${prefix}link_2_l"/>
+        <origin xyz="0.740 0 0" rpy="0 0 0" />
+        <axis xyz="0 1 0" />
+        <limit lower="${radians(-130)}" upper="${radians(80)}" effort="11032.48" velocity="${radians(85)}"/>
+    </joint>
+    <joint name="${prefix}joint_3_u" type="revolute">
+        <parent link="${prefix}link_2_l"/>
+        <child link="${prefix}link_3_u"/>
+        <origin xyz="1.150 0 0" rpy="0 0 0" />
+        <axis xyz="0 -1 0" />
+        <limit lower="${radians(-78.4)}" upper="${radians(78)}" effort="9929.23" velocity="${radians(85)}"/>
+    </joint>
+    <joint name="${prefix}joint_4_r" type="revolute">
+        <parent link="${prefix}link_3_u"/>
+        <child link="${prefix}link_4_r"/>
+        <origin xyz="0.250 0 0" rpy="0 ${radians(90)} 0" />
+        <axis xyz="-1 0 0" />
+        <limit lower="${radians(-180)}" upper="${radians(180)}" effort="2621.3" velocity="${radians(120)}"/>
+    </joint>
+    <joint name="${prefix}joint_5_b" type="revolute">
+        <parent link="${prefix}link_4_r"/>
+        <child link="${prefix}link_5_b"/>
+        <origin xyz="1.225 0 0" rpy="0 0 0" />
+        <axis xyz="0 -1 0" />
+        <limit lower="${radians(-125)}" upper="${radians(180)}" effort="3530.38" velocity="${radians(120)}"/>
+    </joint>
+    <joint name="${prefix}joint_6_t" type="revolute">
+        <parent link="${prefix}link_5_b"/>
+        <child link="${prefix}link_6_t"/>
+        <origin xyz="0.250 0 0" rpy="0 0 0" />
+        <axis xyz="1 0 0" />
+        <limit lower="${radians(-180)}" upper="${radians(180)}" effort="1765.19" velocity="${radians(190)}"/>
+    </joint>
+    <joint name="${prefix}joint_6_t-tool0" type="fixed">
+        <origin xyz="0 0 0" rpy="${radians(180)} ${radians(-90)} 0"/>
+        <parent link="${prefix}link_6_t"/>
+        <child link="${prefix}flange"/>
+    </joint>
```

Comment:
Is this correct? This joint now appears twice.

Reply:
Not supposed to be there twice. I removed the extra joint_6_t-tool0 block, and I pushed to pr which includes the ROS-I maintainer on the list.

Reply:
Side note @gavanderhoorn having this joint in there twice really messes up moveit collision checking when you add a gripper at the end of tool0. Get errors such as gripper_link_1 is in collision with gripper_link_1. All of those errors go away once we take out that redundant joint.

Reply:
It's kind of surprising that the URDF allows the user to do that, I thought links were only allowed to have one parent, but technically since the joint is in there twice it has the same parent.....

Reply:
> when you add a gripper at the end of `tool0`

regardless of the problems caused by duplicated `joint`s and `link`s: you're not supposed to attach grippers / EEF models to `tool0`. That's what `flange` is for.

See also [Coordinate Frames: flange](https://gavanderhoorn.github.io/rep/rep-0199.html#flange).